storage/src/nodestore.rs

```diff
@@ -1318,35 +1318,85 @@ impl NodeStore<Arc<ImmutableProposal>, FileBacked> {
     #[fastrace::trace(short_name = true)]
     #[cfg(feature = "io-uring")]
     pub fn flush_nodes(&self) -> Result<(), FileIoError> {
+        use std::pin::Pin;
+
+        #[derive(Clone, Debug)]
+        struct PinnedBufferEntry {
+            pinned_buffer: Pin<Box<[u8]>>,
+            offset: Option<u64>,
+        }
+
+        /// Helper function to handle completion queue entries and check for errors
+        fn handle_completion_queue(
+            storage: &FileBacked,
+            completion_queue: io_uring::cqueue::CompletionQueue<'_>,
+            saved_pinned_buffers: &mut [PinnedBufferEntry],
+        ) -> Result<(), FileIoError> {
+            for entry in completion_queue {
+                let item = entry.user_data() as usize;
+                let pbe = saved_pinned_buffers
+                    .get_mut(item)
+                    .expect("should be an index into the array");
+
+                if entry.result()
+                    != pbe
+                        .pinned_buffer
+                        .len()
+                        .try_into()
+                        .expect("buffer should be small enough")
```

Comment:
this is a bit annoying. I'm going to suppress this lint. usize -> u64 and u64 -> usize does not truncate on any supported platform.